R/navs-legacy.R

```diff
@@ -439,22 +439,16 @@ findAndMarkSelectedTab <- function(tabs, selected, foundSelected) {
 }
 
 prepTabIcon <- function(x = NULL) {
-  if (is.null(x)) return(NULL)
-  if (!inherits(x, "shiny.tag")) {
-    stop(
-      "`icon` must be a `shiny.tag` object. ",
-      "Try passing `icon()` (or `tags$i()`) to the `icon` parameter.",
-      call. = FALSE
-    )
-  }
 
-  is_fa <- grepl("fa-", tagGetAttribute(x, "class") %||% "", fixed = TRUE)
-  if (!is_fa) {
-    return(x)
+  if (inherits(x, "shiny.tag")) {
+    is_fa <- grepl("fa-", tagGetAttribute(x, "class") %||% "", fixed = TRUE)
+    if (is_fa) {
+      # specify fixed-width for font-awesome
+      x <- tagAppendAttributes(x, class = "fa-fw")
+    }
```

Comment:
This is a small style nit but I have a strong preference for early returns. You could keep some of the early return style of the original function like this:

```suggestion
  if (!inherits(x, "shiny.tag")) return(x)
  
  is_fa <- grepl("fa-", tagGetAttribute(x, "class") %||% "", fixed = TRUE)
  if (!is_fa) return(x)
  
  # specify fixed-width for font-awesome
  tagAppendAttributes(x, class = "fa-fw")
```